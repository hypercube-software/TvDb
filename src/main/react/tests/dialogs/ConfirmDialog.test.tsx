import React from "react";
import { shallow } from "enzyme";

import {
  ConfirmDialog,
  DialogType,
  DialogButton,
} from "../../src/widgets/dialogs/ConfirmDialog";

import { Button } from '@material-ui/core';

describe("ConfirmDialog tests", () => {
  test("Render Error Dialog then click", () => {
    let confimButton = null;  
    const onCloseDialogError = (confirm: DialogButton) => {
        confimButton = confirm;
    };
    const dialog = shallow(
      <ConfirmDialog
        title="Error"
        type={DialogType.ERROR}
        content="The content"
        onResponse={onCloseDialogError}
      />
    );
    dialog.find(Button).simulate("click");
    expect(confimButton).toEqual(DialogButton.OK);
  });

  test("Render Error Dialog", () => {
    const onCloseDialogError = () => {
      //        
    };
    const dialog = shallow(
      <ConfirmDialog
        title="Error"
        type={DialogType.ERROR}
        content="The content"
        onResponse={onCloseDialogError}
      />
    );
    expect(dialog).toMatchSnapshot();
  });

});

