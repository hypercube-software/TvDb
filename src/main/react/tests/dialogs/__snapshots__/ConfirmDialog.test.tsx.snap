// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog tests Render Error Dialog 1`] = `
<WithStyles(ForwardRef(Dialog))
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))>
    Error
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))
    className="dialogContent"
    dividers={true}
  >
    <img
      className="icon"
      src="icons/remove-svgrepo-com.svg"
    />
    <p
      className="message"
    >
      The content
    </p>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="secondary"
      onClick={[Function]}
    >
      Ok
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
